File: EasyMSXPython/EasyMSXSample.py
'''
Created on 22 May 2017

@author: rclegg2
'''
from EasyMSX import EasyMSX


class MyApp:
    
    def __init__(self):

        print "Loading EasyMSX..."
        self.easyMSX = EasyMSX()
    
        print "EasyMSX loaded...."
    
    
        print "\nTeams:"
    
        for t in self.easyMSX.teams:
            print t.name
        
            print "\nBrokers: "
    
        for b in self.easyMSX.brokers:
            print "Broker: " + b.name + "\tAsset Class: " + b.assetClass
            for s in b.strategies:
                print "\tStrategy: " + s.name
                for p in s.parameters:
                    print "\t\tParameter: " + p.name
                    
    
        self.easyMSX.orders.addNotificationHandler(self.processNotification)
        self.easyMSX.routes.addNotificationHandler(self.processNotification)

        self.easyMSX.start()
             
        while True:
            pass
        

    def printOrderBlotter(self):
        print "\nOrder Blotter: \n"
        for o in self.easyMSX.orders:
            print o.field("EMSX_SIDE").value() + "\t" + o.field("EMSX_SEQUENCE").value() + "\t" + o.field("EMSX_STATUS").value() + "  \t" + o.field("EMSX_TICKER").value() + "\t" + o.field("EMSX_AMOUNT").value() + "\t" + o.field("EMSX_TIF").value()
                        
    def printRouteBlotter(self):
        print "\nRoute Blotter: \n"
        for r in self.easyMSX.routes:
            print r.field("EMSX_SEQUENCE").value() + "\t" + r.field("EMSX_ROUTE_ID").value() + "\t" + r.field("EMSX_STATUS").value() + "  \t" + r.field("EMSX_WORKING").value() + "\t" + r.field("EMSX_FILLED").value() + "\t" + r.field("EMSX_AVG_PRICE").value()

    
    def processNotification(self,notification):
        if notification.category == EasyMSX.NotificationCategory.ORDER:
            print "\nChange to Order (" + EasyMSX.NotificationType.asText(notification.type) + "): " + notification.source.field("EMSX_SEQUENCE").value()
            self.printFieldChanges(notification.fieldChanges)
            self.printOrderBlotter()

        elif notification.category == EasyMSX.NotificationCategory.ROUTE:
            print "\nChange to Route (" + EasyMSX.NotificationType.asText(notification.type) + "): " + notification.source.field("EMSX_SEQUENCE").value() + "/" + notification.source.field("EMSX_ROUTE_ID").value()
            self.printFieldChanges(notification.fieldChanges)
            self.printRouteBlotter()
            
        notification.consume=True

    def printFieldChanges(self,fieldChanges):
        for fc in fieldChanges:
            print "Field: " + fc.field.name() + "\tOld: " + fc.oldValue + "\tNew: " + fc.newValue
             
        
if __name__ == '__main__':
    
    myApp = MyApp();
    

File: CSharp/cs_EasyMSXSample-master/EasyMSXSample/EasyMSXSample.cs
using EasyMSX = com.bloomberg.emsx.samples.EasyMSX;
using NotificationHandler = com.bloomberg.emsx.samples.NotificationHandler;
using Log = com.bloomberg.emsx.samples.Log;
using Environment = com.bloomberg.emsx.samples.EasyMSX.Environment;
using Team = com.bloomberg.emsx.samples.Team;
using Broker = com.bloomberg.emsx.samples.Broker;
using BrokerStrategy = com.bloomberg.emsx.samples.BrokerStrategy;
using BrokerStrategyParameter = com.bloomberg.emsx.samples.BrokerStrategyParameter;
using Order = com.bloomberg.emsx.samples.Order;
using Route = com.bloomberg.emsx.samples.Route;
using Notification = com.bloomberg.emsx.samples.Notification;
using NotificationCategory = com.bloomberg.emsx.samples.Notification.NotificationCategory;
using NotificationType = com.bloomberg.emsx.samples.Notification.NotificationType;
using FieldChange = com.bloomberg.emsx.samples.FieldChange;

namespace com.bloomberg.test {

    public class EasyMSXAPISample : NotificationHandler {

	    EasyMSX emsx;
	
        static void Main(string[] args)
        {
            System.Console.WriteLine("Bloomberg - EasyMSX Example - EasyMSXAPISample");

            EasyMSXAPISample example = new EasyMSXAPISample();
            
            example.run(args);
        
            System.Console.ReadKey();    

        }

	    private void run(string[] args) {
	
		    Log.logLevel = Log.LogLevels.DETAILED;

		    System.Console.WriteLine("Initializing EMSXAPI ");

		    //emsx = new EasyMSX(Environment.BETA,"RCAM_API");
		    emsx = new EasyMSX(Environment.BETA);
		    //emsx = new EasyMSX(Environment.BETA, "bpipe-ny-beta.bdns.bloomberg.com", 8194,"CORP\\rclegg2","172.16.21.92");

		
		    System.Console.WriteLine("EMSXAPI initialized OK");
		
		    emsx.orders.addNotificationHandler(this);
		    emsx.routes.addNotificationHandler(this);
		
		    foreach(Team t in emsx.teams) {
			    System.Console.WriteLine("Team: " + t.name);
		    }
	
		    foreach(Broker b in emsx.brokers) {
			    System.Console.WriteLine("Broker: " + b.name);
			
			    System.Console.WriteLine("\tAsset Class: " + b.assetClass.ToString());
			
			    foreach(BrokerStrategy s in b.strategies) {
				    System.Console.WriteLine("\tStrategy: " + s.name);
				
				    foreach(BrokerStrategyParameter p in s.parameters) {
					    System.Console.WriteLine("\t\tParameter: " + p.name);
				    }
			    }
		    }
		
		    System.Console.WriteLine("Existing Orders:");
		
		    foreach(Order o in emsx.orders) {
			    System.Console.WriteLine("\tSequence: " + o.field("EMSX_SEQUENCE").value() + "\tStatus: " + o.field("EMSX_STATUS").value() + "\t Ticker: " + o.field("EMSX_TICKER").value() + "\t Amount: " + o.field("EMSX_AMOUNT").value());
		    }

		    System.Console.WriteLine("Existing Routes:");
		
		    foreach(Route r in emsx.routes) {
			    System.Console.WriteLine("\tSequence: " + r.field("EMSX_SEQUENCE").value() + "\tID: " + r.field("EMSX_ROUTE_ID").value() + "\tStatus: " + r.field("EMSX_STATUS").value()  + "\t Amount: " + r.field("EMSX_AMOUNT").value());
		    }
	    }

	    public void processNotification(Notification notification) {

		    if(notification.category==NotificationCategory.ORDER) {
			    if(notification.type.Equals(NotificationType.ERROR)) {
				    System.Console.WriteLine("Order Notification [" + notification.category.ToString() + "|" + notification.type.ToString() + "] "  + " Error Code:" + notification.errorCode() + "\t" + notification.errorMessage());
			    } else {
				    System.Console.WriteLine("Order Notification [" + notification.category.ToString() + "|" + notification.type.ToString() + "] "  + " Order: " + notification.getOrder().field("EMSX_SEQUENCE").value() + " : No. of affected fields: " + notification.getFieldChanges().Count);
				    notification.consume = true;
				    foreach(FieldChange fc in notification.getFieldChanges()) {
					    //System.Console.WriteLine("\t\tChange: " + fc.field.name() + "\tOld Value: " + fc.oldValue + "\tNew Value: " + fc.newValue);
				    }
			    }
		    } 
		    else if(notification.category==NotificationCategory.ROUTE) {
			    if(notification.type.Equals(NotificationType.ERROR)) {
				    System.Console.WriteLine("Route Notification [" + notification.category.ToString() + "|" + notification.type.ToString() + "] "  + " Error Code:" + notification.errorCode() + "\t" + notification.errorMessage());
			    } else {
				    System.Console.WriteLine("Route Notification [" + notification.category.ToString() + "|" + notification.type.ToString() + "] "  + " Route: " + notification.getRoute().field("EMSX_SEQUENCE").value() + "." + notification.getRoute().field("EMSX_ROUTE_ID").value() + " : No. of affected fields: " + notification.getFieldChanges().Count);
				    notification.consume = true;
				    foreach(FieldChange fc in notification.getFieldChanges()) {
					    System.Console.WriteLine("\t\tChange: " + fc.field.name() + "\tOld Value: " + fc.oldValue + "\tNew Value: " + fc.newValue);
				    }
			    }
		    }
	    }
    }
}